src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageService`IVsLanguageDebugInfo.cs

```diff
@@ -33,12 +50,40 @@ public int GetProximityExpressions(IVsTextBuffer pBuffer, int iLine, int iCol, i
 
         public int ValidateBreakpointLocation(IVsTextBuffer pBuffer, int iLine, int iCol, TextSpan[] pCodeSpan)
         {
+            var textBuffer = _editorAdaptersFactory.GetDataBuffer(pBuffer);
+            if (textBuffer == null)
+            {
+                // Can't resolve the text buffer, let someone else deal with this breakpoint.
+                return VSConstants.E_NOTIMPL;
+            }
+
+            var snapshot = textBuffer.CurrentSnapshot;
+            if (!ValidBreakpointLocation(snapshot, iLine, iCol))
+            {
+                // The point disappeared between sessions. Do not allow a breakpoint here.
+                return VSConstants.E_FAIL;
+            }
+
             var dialogResult = _waitDialogFactory.TryCreateWaitDialog(
                 title: "Determining breakpoint location...",
                 message: "Razor Debugger",
-                (context) =>
+                async (context) =>
                 {
-                    return Task.FromResult(VSConstants.E_NOTIMPL);
+                    var breakpointRange = await _breakpointResolver.TryResolveBreakpointRangeAsync(textBuffer, iLine, iCol, context.CancellationToken);
+                    if (breakpointRange == null)
+                    {
+                        // No applicable breakpoint location.
+                        return VSConstants.E_FAIL;
```

Comment:
How does VSClientLanguageServer handle these error-code? I feel like it might be prudent to have this class take a logger and log some extra info about these edge cases, if not in this PR then as an issue filed for future improvement.

Reply:
> How does VSClientLanguageServer handle these error-code?

It doesn't. The consumer of these are actually at a native VS-Platform layer. In this case these aren't error codes. They're just a descriptive return value 😄.

For this method there's really 3 options:
* S_OK - Breakpoint can be set
* E_FAIL - Breakpoint should not be able to be set
* E_NOTIMPL - Let another debug info provider try